deepchem/data/dft_data.py

```diff
@@ -0,0 +1,319 @@
+"""
+Density Functional Theory Data
+Derived from: https://github.com/mfkasim1/xcnn/blob/f2cb9777da2961ac553f256ecdcca3e314a538ca/xcdnn2/entry.py
+"""
+from __future__ import annotations
+import os
+from abc import abstractmethod, abstractproperty
+from typing import List, Dict, Optional, Union
+import numpy as np
+import torch
+try:
+    import dqc
+    from dqc.system.mol import Mol
+    from dqc.system.base_system import BaseSystem
+    from dqc.grid.base_grid import BaseGrid
+    from deepchem.utils.dftutils import KSCalc
+except ModuleNotFoundError:
+    raise ModuleNotFoundError("This utility requires dqc")
+
+import yaml
+from yaml.loader import SafeLoader
+
+
+class DFTSystem(dict):
+    """
+    The DFTSystem class creates and returns the various systems in an entry object as dictionaries. 
+
+    References
+    ----------
+    Kasim, Muhammad F., and Sam M. Vinko. "Learning the exchange-correlation functional from nature with fully differentiable density functional theory." Physical Review Letters 127.12 (2021): 126403.
+    
+    https://github.com/diffqc/dqc/blob/0fe821fc92cb3457fb14f6dff0c223641c514ddb/dqc/system/base_system.py
+    """
+
+    created_systems: Dict[str, DFTSystem] = {}
+
+    @classmethod
+    def create(cls, system: Dict) -> DFTSystem:
+        # create the system if it has not been created
+        # otherwise, return the previously created system
+
+        system_str = str(system)
+        if system_str not in cls.created_systems:
+            cls.created_systems[system_str] = DFTSystem(system)
+        return cls.created_systems[system_str]
+
+    def __init__(self, system: Dict):
+        super().__init__(system)
+
+        # caches
+        self._caches = {}
+
+    def get_dqc_system(self, pos_reqgrad: bool = False) -> BaseSystem:
```

Comment:
It feels like `get_dqc_system` is the wrong name. How about `get_dft_mol` instead? The method returns `Mol` object.